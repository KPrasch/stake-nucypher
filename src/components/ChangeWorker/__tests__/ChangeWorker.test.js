import React from 'react';
import ChangeWorker from '../ChangeWorker';
import { render, fireEvent, wait } from '@testing-library/react'

describe('<ChangeWorker />', () => {

  it('should render correctly', () => {
    const { asFragment } = render(<ChangeWorker />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger onChangeWorker on providing worker', async () => {
    const onChangeWorker = jest.fn(() => {});
    const { getByLabelText, getByText } = render(<ChangeWorker onChangeWorker={onChangeWorker} />);

    await wait(() => {
      fireEvent.change(getByLabelText('Worker'), { target: { value: '0x6453ee19CDa53491Be3772Ff33B499b3A3e47886' } });
      fireEvent.click(getByText('Set worker'));
    });

    expect(onChangeWorker).toBeCalledWith(expect.objectContaining({ workerAddress: '0x6453ee19CDa53491Be3772Ff33B499b3A3e47886' }));
  });

  it('should validate input from user correctly', async () => {
    const { getByLabelText, getByText, asFragment } = render(<ChangeWorker />);

    await wait(() => {
      fireEvent.change(getByLabelText('Worker'), { target: { value: '0x1111' } });
      fireEvent.click(getByText('Set worker'));
    });

    expect(asFragment()).toMatchSnapshot();

    await wait(() => {
      fireEvent.change(getByLabelText('Worker'), { target: { value: '' } });
      fireEvent.click(getByText('Set worker'));
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
