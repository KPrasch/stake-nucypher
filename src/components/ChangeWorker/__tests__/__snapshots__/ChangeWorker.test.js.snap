// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangeWorker /> should render correctly 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-sm-2"
        for="worker-address"
      >
        Worker
      </label>
      <div
        class="col-sm-10"
      >
        <input
          class="form-control"
          id="worker-address"
          name="workerAddress"
          type="text"
          value=""
        />
        <div
          class="invalid-feedback"
          style="display: none;"
        />
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <button
        class="btn btn-primary"
        type="submit"
      >
        Set worker
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<ChangeWorker /> should validate input from user correctly 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-sm-2"
        for="worker-address"
      >
        Worker
      </label>
      <div
        class="col-sm-10"
      >
        <input
          class="form-control"
          id="worker-address"
          name="workerAddress"
          type="text"
          value="0x1111"
        />
        <div
          class="invalid-feedback"
          style="display: inline;"
        >
          Should be a valid Ethereum address
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <button
        class="btn btn-primary"
        disabled=""
        type="submit"
      >
        Set worker
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<ChangeWorker /> should validate input from user correctly 2`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-sm-2"
        for="worker-address"
      >
        Worker
      </label>
      <div
        class="col-sm-10"
      >
        <input
          class="form-control"
          id="worker-address"
          name="workerAddress"
          type="text"
          value=""
        />
        <div
          class="invalid-feedback"
          style="display: inline;"
        >
          Field is required.
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <button
        class="btn btn-primary"
        disabled=""
        type="submit"
      >
        Set worker
      </button>
    </div>
  </form>
</DocumentFragment>
`;
